src/waterfall/polarplot.h

```diff
@@ -0,0 +1,125 @@
+#pragma once
+
+#include <QQuickPaintedItem>
+#include <QImage>
+
+#include "logger.h"
+#include "ringvector.h"
+#include "waterfall.h"
+#include "waterfallgradient.h"
+
+Q_DECLARE_LOGGING_CATEGORY(waterfall)
+
+/**
+ * @brief Polar widget
+ *
+ */
+class PolarPlot : public Waterfall
+{
+    Q_OBJECT
+
+public:
+    /**
+     * @brief Construct a new PolarPlot object
+     *
+     * @param parent
+     */
+    PolarPlot(QQuickItem *parent = nullptr);
+
+    /**
+     * @brief This is used by the qml paint event
+     *  This paint the a polar waterfall in qml
+     *
+     * @param painter
+     */
+    void paint(QPainter *painter) final override;
+
+    /**
+     * @brief Set the polar Image
+     *
+     * @param image
+     */
+    void setImage(const QImage &image);
+
+    /**
+     * @brief Draw a list of points in the waterfall
+     *
+     * @param points
+     * @param confidence
+     * @param initPoint
+     * @param length
+     * @param distance
+     */
+    Q_INVOKABLE void draw(const QVector<double>& points, float confidence = 0, float initPoint = 0, float length = 50,
```

Comment:
@patrickelectric you intend to rework the parameters here correct? (perhaps add a todo in those cases)

Reply:
Yes